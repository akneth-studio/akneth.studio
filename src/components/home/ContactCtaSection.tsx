'use client'

import CTAButton from "../CTAButton";

const calendar = process.env.NEXT_PUBLIC_CALENDAR;

export default function ContactCtaSection() {
  return (
    <section className="contact-cta text-center bg-light py-5 px-4">
      <h2>Kontakt</h2>
      <p>
        Masz pytania lub chcesz zlecić projekt? <br />
        Umów się na bezpłatną konsultację lub napisz przez formularz kontaktowy.
      </p>
      <div className="d-flex justify-content-center gap-3 mt-3">
        <CTAButton
          type='button'
          text="Formularz kontaktowy"
          variant="primary"
          to="/contact"
        />
        <CTAButton
          type='button'
          text="Harmonogram spotkań"
          variant="outline-primary"
          onClick={() => window.open(calendar, "_blank")}
        />
      </div>
    </section>
  );
}
