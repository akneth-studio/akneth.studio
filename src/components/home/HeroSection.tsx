import CTAButton from "../CTAButton";

export default function HeroSection() {
  return (
    <section className="hero text-center py-4">
      <div id='lead-gradient' className='px-4 mb-4'>
        <h1>Szablony, automatyzacje i indywidualne rozwiązania<br/>
        dla firm i klientów indywidualnych.</h1><br />
        <p className='lead fs-1'>Elastyczność, uczciwość i szybka realizacja - Twoje potrzeby, mój priorytet.</p>
      </div>
      <CTAButton
        type='button'
        text="Skontaktuj się"
        variant="primary"
        to="/contact"
        size='lg'
      />
    </section>
  );
}
