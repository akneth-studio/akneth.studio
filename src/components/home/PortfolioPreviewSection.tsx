export default function PortfolioPreviewSection() {
  return (
    <section className="portfolio-preview container bg-light py-5 px-4">
      <h2>Portfolio</h2>
      <ul className="list-unstyled">
        <li>
          <strong>Szablon Google Sheets dla kawiarni</strong> - <a href="https://drive.google.com/drive/folders/1NySz1o6OOxVM-yWps897LPPegkeOoqLG?usp=sharing" target="_blank" rel="noopener noreferrer">zobacz na Google Drive</a>
        </li>
        <li>
          <strong>Strona pisarska</strong> - <a href="https://github.com/reisene/wilczar.writer" target="_blank" rel="noopener noreferrer">repozytorium na GitHub</a>
        </li>
        <li>
          <strong>Serwis hulajnóg (prototyp)</strong> - <a href="https://github.com/reisene/HulajDusza-serwis" target="_blank" rel="noopener noreferrer">repozytorium na GitHub</a>
        </li>
      </ul>
      <p className="mt-3">Wkrótce więcej przykładów!</p>
    </section>
  );
}
