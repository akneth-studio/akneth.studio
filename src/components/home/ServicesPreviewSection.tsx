import CTAButton from "../CTAButton";

export default function ServicesPreviewSection() {
  return (
    <section className="services-preview bg-light py-5 px-4">
      <div className="container">
        <h2>Usługi</h2>
        <ul className="list-unstyled">
          <li><strong>Szablony Google</strong> - projektowanie funkcjonalnych arkuszy dla firm i osób prywatnych.</li>
          <li><strong>Automatyzacje</strong> - usprawnianie codziennej pracy dzięki prostym, praktycznym rozwiązaniom.</li>
        </ul>
        <CTAButton
          type='button'
          text="Zobacz pełną ofertę"
          variant="outline-primary"
          to="/services"
        />
      </div>
    </section>
  );
}
