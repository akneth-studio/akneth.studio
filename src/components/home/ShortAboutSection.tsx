import CTAButton from "../CTAButton";

export default function ShortAboutSection() {
  return (
    <section className="about-preview container bg-light py-5 px-4">
      <h2>O mnie</h2>
      <p>
        Nazywam się Katarzyna Pawłowska-Malesa. Jestem osobą otwartą, elastyczną i nie boję się żadnych wyzwań - zarówno w życiu, jak i w pracy. Programuję, tworzę szablony i automatyzacje, a każdemu projektowi poświęcam maksimum uwagi. Wierzę, że indywidualne podejście i szczerość to podstawa dobrej współpracy.
      </p>
      <CTAButton
        type='button'
        text="Dowiedz się więcej"
        variant="outline-primary"
        to="/about"
      />
    </section>
  );
}
