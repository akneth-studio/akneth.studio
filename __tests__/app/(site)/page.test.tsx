/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import HomePage from '@/app/(site)/page';
import HeroSection from '@/components/home/HeroSection';
import ShortAboutSection from '@/components/home/ShortAboutSection';
import ServicesPreviewSection from '@/components/home/ServicesPreviewSection';
import PortfolioPreviewSection from '@/components/home/PortfolioPreviewSection';
import ContactCtaSection from '@/components/home/ContactCtaSection';

// Mock the components
jest.mock('@/components/home/HeroSection', () => {
  return jest.fn(() => <div data-testid="hero-section">Hero Section</div>);
});

jest.mock('@/components/home/ShortAboutSection', () => {
  return jest.fn(() => <div data-testid="short-about-section">Short About Section</div>);
});

jest.mock('@/components/home/ServicesPreviewSection', () => {
  return jest.fn(() => <div data-testid="services-preview-section">Services Preview Section</div>);
});

jest.mock('@/components/home/PortfolioPreviewSection', () => {
  return jest.fn(() => <div data-testid="portfolio-preview-section">Portfolio Preview Section</div>);
});

jest.mock('@/components/home/ContactCtaSection', () => {
  return jest.fn(() => <div data-testid="contact-cta-section">Contact CTA Section</div>);
});

// Mock environment variables
process.env.NEXT_PUBLIC_SITE_URL = require('../../../jest.setup').NEXT_PUBLIC_SITE_URL || 'https://akneth.pl';

describe('HomePage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render all sections correctly', () => {
    render(<HomePage />);
    
    // Check if all sections are rendered
    expect(screen.getByTestId('hero-section')).toBeInTheDocument();
    expect(screen.getByTestId('short-about-section')).toBeInTheDocument();
    expect(screen.getByTestId('services-preview-section')).toBeInTheDocument();
    expect(screen.getByTestId('portfolio-preview-section')).toBeInTheDocument();
    expect(screen.getByTestId('contact-cta-section')).toBeInTheDocument();
  });

  it('should render the correct layout structure', () => {
    render(<HomePage />);
    
    // Check if the main container exists
    const container = screen.getByTestId('hero-section').parentElement?.parentElement;
    expect(container).toBeInTheDocument();
    
    // Check if the row structure exists (it should have gy-5 class based on the component)
    const row = screen.getByTestId('services-preview-section').parentElement?.parentElement?.parentElement;
    expect(row).toBeInTheDocument();
    expect(row).toHaveClass('gy-5');
    
    // Check if the columns exist
    const servicesColumn = screen.getByTestId('services-preview-section').parentElement;
    expect(servicesColumn).toBeInTheDocument();
    expect(servicesColumn).toHaveClass('col-12', 'col-lg-6');
    
    const portfolioColumn = screen.getByTestId('portfolio-preview-section').parentElement;
    expect(portfolioColumn).toBeInTheDocument();
    expect(portfolioColumn).toHaveClass('col-12', 'col-lg-6');
  });

  it('should have correct metadata', () => {
    // Check if metadata is defined
    const { metadata } = require('@/app/(site)/page');
    expect(metadata).toBeDefined();
    expect(metadata.title).toEqual({
      default: 'Strona główna',
      template: '% | AKNETH Studio',
    });
    expect(metadata.description).toBe('Witamy na stronie głównej naszej witryny.');
    expect(metadata.keywords).toEqual(['strona główna', 'AKNETH Studio', 'portfolio']);
    expect(metadata.alternates?.canonical).toBe('https://akneth.pl');
  });

  it('should match snapshot', () => {
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });
});
