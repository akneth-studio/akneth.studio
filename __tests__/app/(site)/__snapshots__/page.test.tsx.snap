// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomePage should match snapshot 1`] = `
<div>
  <div
    data-testid="hero-section"
  >
    Hero Section
  </div>
  <div
    data-testid="short-about-section"
  >
    Short About Section
  </div>
  <div
    class="gy-5"
  >
    <div
      class="row gy-md-5 gy-0 gx-5 align-items-center"
    >
      <div
        class="col-12 col-lg-6"
      >
        <div
          data-testid="services-preview-section"
        >
          Services Preview Section
        </div>
      </div>
      <div
        class="col-12 col-lg-6"
      >
        <div
          data-testid="portfolio-preview-section"
        >
          Portfolio Preview Section
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="contact-cta-section"
  >
    Contact CTA Section
  </div>
</div>
`;
